packages/client/src/__tests__/integration/happy/interactive-transactions-postgres/test.ts

```diff
@@ -0,0 +1,473 @@
+    })
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(
+      `Transaction API error: Transaction already closed: Transaction is no longer valid. Last state: 'Expired'.`,
+    )
+
+    expect(await prisma.user.findMany()).toHaveLength(0)
+  })
+
+  /**
+   * Transactions should fail if they time out on `timeout`
+   */
+  test('timeout override', async () => {
+    const result = prisma.$transaction(
+      async (prisma) => {
+        await prisma.user.create({
+          data: {
+            email: 'user_1@website.com',
+          },
+        })
+
+        await new Promise((res) => setTimeout(res, 6000))
+      },
+      {
+        maxWait: 200,
+        timeout: 500,
+      },
+    )
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(
+      `Transaction API error: Transaction already closed: Transaction is no longer valid. Last state: 'Expired'.`,
+    )
+
+    expect(await prisma.user.findMany()).toHaveLength(0)
+  })
+
+  /**
+   * Transactions should fail if they time out on `maxWait`
+   */
+  test.skip('maxWait override', async () => {
+    const result = prisma.$transaction(
+      async (prisma) => {
+        await prisma.user.create({
+          data: {
+            email: 'user_1@website.com',
+          },
+        })
+
+        await new Promise((res) => setTimeout(res, 5))
+
+        return prisma.user.findMany()
+      },
+      {
+        maxWait: 0,
+      },
+    )
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+                
+                            Invalid \`prisma.user.findMany()\` invocation:
+                
+                
+                              Transaction API error: Transaction already closed: Transaction is no longer valid. Last state: 'Expired'.
+                        `)
+  })
+
+  /**
+   * Transactions should fail and rollback if thrown within
+   */
+  test('rollback throw', async () => {
+    const result = prisma.$transaction(async (prisma) => {
+      await prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      })
+
+      throw new Error('you better rollback now')
+    })
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`you better rollback now`)
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(0)
+  })
+
+  /**
+   * A transaction might fail if it's called inside another transaction
+   * //! this does not behave the same for all dbs (sqlite)
+   */
+  test('nested create', async () => {
+    const result = prisma.$transaction(async (tx) => {
+      await tx.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      })
+
+      await prisma.$transaction(
+        async (prisma) => {
+          await prisma.user.create({
+            data: {
+              email: 'user_2@website.com',
+            },
+          })
+        },
+        {
+          timeout: 1000,
```

Comment:
Why is the explicit timeout necessary here?

Reply:
I don't think so.